### PR TITLE
feat(compute): add compute reservation create shared

### PR DESCRIPTION
**Pull request by TetyanaYahodska**
## Description

Fixes #<ISSUE-NUMBER>

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [x] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [ ] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  // Creates a shared reservation with the given name in the given zone.
+  public void createSharedReservation(
+      String projectId, String zone,
+      String reservationName, String instanceTemplateUri, int vmCount)
+      throws ExecutionException, InterruptedException, TimeoutException {
+
+    ShareSettings shareSettings = ShareSettings.newBuilder()
+        .setShareType(String.valueOf(ShareSettings.ShareType.SPECIFIC_PROJECTS))
+        .putProjectMap("CONSUMER_PROJECT_ID_1", ShareSettingsProjectConfig.newBuilder().build())
+        .putProjectMap("CONSUMER_PROJECT_ID_2", ShareSettingsProjectConfig.newBuilder().build())
```
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java:72]: In the test envaironment we don't have possibility to create additional projects to implement this feature. For this reason we have different implementation for this sample. And for test we mocked everything.

File: compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    // Name of the reservation to be created.
+    String reservationName = "YOUR_RESERVATION_NAME";
+    // The URI of the global instance template to be used for creating the reservation.
+    String instanceTemplateUri = String.format(
+        "projects/%s/global/instanceTemplates/YOUR_INSTANCE_TEMPLATE_NAME", projectId);
+    // Number of instances for which capacity needs to be reserved.
+    int vmCount = 3;
+    // In your main method, create a real ReservationsClient
+    ReservationsClient client = ReservationsClient.create();
+    // Create an instance of your class, passing in the real client
```
[the remark this answers is not in the document]
Comment (TetyanaYahodska) [reply compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java:57]: Thank you!

File: compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (m-strzelczyk) [reply compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java:72]: Please indicate, those CONSUMER_PROJECT_ID_1 and _2 are the projects that will have access to the reservation.

File: compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+  public CreateSharedReservation(ReservationsClient reservationsClient) {
+    this.reservationsClient = reservationsClient;
+  }
+
+  public static void main(String[] args)
+      throws IOException, ExecutionException, InterruptedException, TimeoutException {
+    // TODO(developer): Replace these variables before running the sample.
+    // The ID of the project where you want to reserve resources
+    // and where the instance template exists.
+    String projectId = "YOUR_PROJECT_ID";
```
Comment (m-strzelczyk) [inline compute/cloud-client/src/main/java/compute/reservation/CreateSharedReservation.java:45]: Make sure to highlight the fact, that this project MUST be allowlisted in `compute.sharedReservationsOwnerProjects` org policy. See more at: https://cloud.google.com/compute/docs/instances/reservations-shared#shared_reservation_constraint

File: compute/cloud-client/src/test/java/compute/reservation/ReservationIT.java [hunk 4]
```diff
@@ -177,4 +223,50 @@ public void testCreateReservationWithRegionInstanceTemplate()
...
+
+    OperationFuture mockFuture = mock(OperationFuture.class);
+    when(mockReservationsClient.insertAsync(PROJECT_ID, ZONE, reservation))
+        .thenReturn(mockFuture);
+
+    // Mock the Operation
+    Operation mockOperation = mock(Operation.class);
+    when(mockFuture.get(3, TimeUnit.MINUTES)).thenReturn(mockOperation);
+    when(mockOperation.hasError()).thenReturn(false);
+    when(mockOperation.getStatus()).thenReturn(Operation.Status.DONE);
```
Comment (m-strzelczyk) [inline compute/cloud-client/src/test/java/compute/reservation/ReservationIT.java:260]: Add explenation why we need to mock + what would have to be changes if someone wants to test this code without mocking.

**Review by m-strzelczyk — CHANGES_REQUESTED** [review]

**Review by Sita04 — APPROVED** [review]
One nit. LGTM.
Kindly get Maciej's LGTM before merging.

**Review by m-strzelczyk — APPROVED** [review]

**Pull request merged**